File: Loenn/entities/rust_berry.lua
local drawableSprite = require("structs.drawable_sprite")
local utils = require("utils")

local rustBerry = {}

rustBerry.name = "FactoryHelper/RustBerry"

rustBerry.placements = {
    name = "rust_berry"
}

function rustBerry.sprite(room, entity)
    local button = drawableSprite.fromTexture("objects/FactoryHelper/rustBerry/berry_01", entity)
    local case = drawableSprite.fromTexture("objects/FactoryHelper/rustBerry/gear_01", entity)

    return {
        case,
        button
    }
end

function rustBerry.selection(room, entity)
    return utils.rectangle(entity.x - 8, entity.y - 7, 16, 14)
end

return rustBerry

File: Ahorn/entities/rustBerry.jl
module FactoryHelperRustBerry

using ..Ahorn, Maple

@mapdef Entity "FactoryHelper/RustBerry" RustBerry(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
    "Rusty Berry (FactoryHelper)" => Ahorn.EntityPlacement(
        RustBerry,
    )
)

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::RustBerry, room::Maple.Room)
    Ahorn.drawSprite(ctx, "objects/FactoryHelper/rustBerry/berry_01", 0, 0)
    Ahorn.drawSprite(ctx, "objects/FactoryHelper/rustBerry/gear_01", 0, 0)
end

function Ahorn.selection(entity::RustBerry)
    x, y = Ahorn.position(entity)

    return Ahorn.Rectangle(x - 4, y - 6, 8, 12)
end

end
